File: mobile/ios/LiveRecallGlasses/LiveRecallGlasses/Views/PovPreview.swift
// PovPreview.swift
//
// 16:9 viewport that hosts the LiveKit VideoView for whichever local video
// track is currently published — glasses POV via BufferCapturer or the
// phone camera fallback. Visual treatment matches the .pov-frame block in
// phone/glasses.html (purple border, dashed reticle, "POV · first-person"
// chip in the top-left).

import SwiftUI
import LiveKit

struct PovPreview: View {
    let track: LocalVideoTrack?

    var body: some View {
        ZStack(alignment: .topLeading) {
            // Black backdrop so a missing track still draws the right shape.
            Theme.background

            if let track {
                // SwiftUIVideoView ships with the LiveKit Swift SDK and
                // auto-attaches/detaches the underlying renderer.
                SwiftUIVideoView(track)
                    .clipShape(RoundedRectangle(cornerRadius: 16))
            } else {
                VStack(spacing: 6) {
                    Text("waiting for video")
                        .font(.system(size: 13, weight: .medium))
                        .foregroundColor(Theme.muted)
                    Text("connect to start streaming")
                        .font(.system(size: 11))
                        .foregroundColor(Theme.muted.opacity(0.7))
                }
                .frame(maxWidth: .infinity, maxHeight: .infinity)
            }

            // Reticle: dashed inset hint that this is "first-person".
            RoundedRectangle(cornerRadius: 12)
                .strokeBorder(style: StrokeStyle(lineWidth: 1, dash: [4, 4]))
                .foregroundColor(Theme.accent.opacity(0.25))
                .padding(8)

            Text("POV · FIRST-PERSON")
                .font(.system(size: 10, weight: .semibold))
                .tracking(0.8)
                .padding(.horizontal, 8)
                .padding(.vertical, 4)
                .background(Theme.background.opacity(0.6))
                .overlay(
                    RoundedRectangle(cornerRadius: 6)
                        .stroke(Theme.accent.opacity(0.35), lineWidth: 1)
                )
                .clipShape(RoundedRectangle(cornerRadius: 6))
                .foregroundColor(Color(hex: 0xDDD6FE))
                .padding(10)
        }
        .aspectRatio(16/9, contentMode: .fit)
        .clipShape(RoundedRectangle(cornerRadius: 16))
        .overlay(
            RoundedRectangle(cornerRadius: 16)
                .stroke(Theme.accent.opacity(0.3), lineWidth: 1)
        )
    }
}

File: mobile/android/LiveRecallGlasses/app/src/main/kotlin/com/liverecall/glasses/ui/components/PovPreview.kt
package com.liverecall.glasses.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.aspectRatio
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.liverecall.glasses.ui.theme.Tokens
import io.livekit.android.compose.VideoTrackView
import io.livekit.android.room.track.LocalVideoTrack

/**
 * 16:9 viewport with the same purple accent treatment as the .pov-frame
 * block in phone/glasses.html. Renders the LiveKit local video track when
 * we have one (glasses stream or phone fallback both look identical from
 * here — same VideoTrackView).
 *
 * NOTE on the renderer: livekit-android-compose-components ships
 * `VideoTrackView` taking a `Track`. If your version pins a different
 * composable name (older builds export `VideoRenderer`), swap the call
 * below — the rest of the file is renderer-agnostic.
 */
@Composable
fun PovPreview(track: LocalVideoTrack?) {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .aspectRatio(16f / 9f)
            .clip(RoundedCornerShape(16.dp))
            .background(Tokens.Background)
            .border(1.dp, Tokens.Accent.copy(alpha = 0.30f), RoundedCornerShape(16.dp)),
    ) {
        if (track != null) {
            VideoTrackView(
                videoTrack = track,
                modifier = Modifier.fillMaxSize(),
            )
        } else {
            Column(
                modifier = Modifier.fillMaxSize(),
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.Center,
            ) {
                Text("waiting for video", color = Tokens.Muted, fontSize = 13.sp,
                     fontWeight = FontWeight.Medium)
                Text("connect to start streaming", color = Tokens.Muted.copy(alpha = 0.7f),
                     fontSize = 11.sp)
            }
        }
        Box(
            modifier = Modifier
                .fillMaxSize()
                .padding(8.dp)
                .border(1.dp, Tokens.Accent.copy(alpha = 0.25f), RoundedCornerShape(12.dp)),
        )
        Text(
            "POV · FIRST-PERSON",
            color = androidx.compose.ui.graphics.Color(0xFFDDD6FE),
            fontSize = 10.sp,
            fontWeight = FontWeight.SemiBold,
            modifier = Modifier
                .padding(10.dp)
                .clip(RoundedCornerShape(6.dp))
                .background(Tokens.Background.copy(alpha = 0.6f))
                .border(1.dp, Tokens.Accent.copy(alpha = 0.35f), RoundedCornerShape(6.dp))
                .padding(horizontal = 8.dp, vertical = 4.dp),
        )
    }
}
